drivers/serial/Kconfig.stm32

```diff
@@ -25,6 +25,13 @@ config UART_1
 	  Enable support for USART1 port in the driver.
 	  Say y here if you want to use USART1 device.
 
+config UART_1_HW_FC
```

Comment:
Please check how we could  introduce dtscts support as a device tree property (uart-has-rtscts) instead of Kconfig.